gen/utils/data/data.go

```diff
@@ -0,0 +1,144 @@
+package data
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/CiscoDevNet/terraform-provider-aci/v2/gen/utils"
+	"github.com/CiscoDevNet/terraform-provider-aci/v2/gen/utils/logger"
+)
+
+// Initialize a logger instance for the generator.
+var genLogger = logger.InitalizeLogger()
+
+type DataStore struct {
+	// A map containing all the information about the classes required to render the templates.
+	Classes map[string]Class
+	// The host from which the meta data is retrieved.
+	metaHost string
+	// A list of all the classes that have been retrieved from the remote location.
+	// This is used to avoid retrieving the same class multiple times.
+	retrievedClasses []string
+}
+
+func NewDataStore() *DataStore {
+	dataStore := &DataStore{Classes: make(map[string]Class)}
+	// Set the meta data host for retrieval of meta files.
+	dataStore.setMetaHost()
+	// Check if classes are set in the environment variable 'GEN_ACI_TF_META_CLASSES' and retrieve the meta files for those classes.
+	dataStore.retrieveEnvMetaClassesFromRemote()
+	// Refresh the meta files from the remote location if specified in the environment variable 'GEN_ACI_TF_META_REFRESH'.
```

Comment:
Probably need to add this here:
```
func NewDataStore(className){
dataStore.retrieveMetaFileFromRemote(className)
```

Reply:
Can you elaborate what you mean, because I do not understand your comment

Reply:
You can ignore this, I thought we needed to call the method inside here